<?xml version="1.0"?>
<launch>

    <!-- General Settings -->
        <!-- Debug Parameters -->
        <arg name="debug"           default="false"/>
        <arg unless="$(arg debug)"  name="debug_config" value="" />
        <arg if=    "$(arg debug)"  name="debug_config" value="xterm -e gdb --ex run --args" />
        <arg name="debug_string"      value="xterm -e gdb -ex run -args" if="$(arg debug)"/>
        <arg name="debug_string"      value="" unless="$(arg debug)"/>

    <!-- ############################## -->
    <!--    Parameter Loading Section   -->
    <!-- ############################## -->

        <!-- System Settings -->
        <!-- (Load global settings and parameters of system to parameter server) -->
        <rosparam command="load" file="$(find robot_toolbox)/test/test_params.yaml" />

    <!-- ############################## -->
    <!--        Node Launching          -->
    <!-- ############################## -->

        <!-- User-Frame Test-Node -->
        <node name="test_param" pkg="robot_toolbox" type="test_param" output="screen" launch-prefix="$(arg debug_string)"/>
</launch>